examples/gbasf2_fei_training/fei_grid_workflow.py

```diff
@@ -0,0 +1,473 @@
+                outputs[inname] += glob.glob(os.path.join(folder, "*.root"))
+
+        with open(self.get_output_file_name("list_of_output_directories.json"), 'w') as jsonfile:
+            json.dump(outputs, jsonfile, sort_keys=True, indent=2)
+
+
+class FEIAnalysisTask(Basf2PathTask):
+
+    batch_system = "gbasf2"
+
+    git_hash = luigi.Parameter(hashed=True, default=get_basf2_git_hash(), significant=False)
+
+    gbasf2_project_name_prefix = luigi.Parameter(significant=False)
+    gbasf2_input_dslist = luigi.Parameter(hashed=True, significant=False)
+
+    cache = luigi.IntParameter(significant=False)
+    monitor = luigi.BoolParameter(significant=False)
+    stage = luigi.IntParameter()
+    mode = luigi.Parameter()
+
+    def output(self):
+
+        for outname in fei_analysis_outputs[self.stage]:
+            yield self.add_to_output(outname)
+
+    def requires(self):
+
+        if self.stage == -1:
+
+            return []  # default implementation, as in the luigi.Task (if no requirements are present)
+
+        else:
+
+            # need the timestamp of the additional inputs to build their TMP-SE path
+            yield PrepareInputsTask(
+                mode="AnalysisInput",
+                stage=self.stage - 1,
+                remote_tmp_directory=luigi.get_setting("remote_tmp_directory"),
+                remote_initial_se=luigi.get_setting("remote_initial_se"),
+            )
+
+            # need a symlink to the merged mcParticlesCount.root file
+            yield MergeOutputsTask(
+                mode="Merging",
+                stage=-1,
+                ncpus=luigi.get_setting("local_cpus"),
+            )
+
+            # need symlinks to *.xml files of FEI training of previous stages
+            for fei_stage in range(self.stage):
+
+                yield FEITrainingTask(
+                    mode="Training",
+                    stage=fei_stage,
+                )
+
+    def create_path(self):
+
+        luigi.set_setting("gbasf2_cputime", grid_cpu_time[self.stage])
+
+        # determine the remote TMP-SE destination of input tarball for gbasf2 command
+        if self.stage > -1:
+            timestamp = open(f"{self.get_input_file_names('successfull_input_upload.txt')[0]}", "r").read().strip()
+            additional_file = os.path.join(luigi.get_setting("remote_tmp_directory").rstrip('/')+timestamp,
+                                           "stage"+str(self.stage - 1), "sub00", "fei_analysis_inputs.tar.gz")
+            luigi.set_setting("gbasf2_input_datafiles", [additional_file])
+
+            # create symlinks to files, which are needed for current FEI analysis stage
+            for key in self.get_input_file_names():
+                if key.endswith(".xml"):
+                    filepath = '"' + self.get_input_file_names(key)[0] + '"'
+                    adjusted_key = '"' + key + '"'
+                    os.system(f"ln -sf {filepath} {adjusted_key}")
```

Comment:
Replace with `os.symlink` and if you want to force it, you can use the following function, which I found by googling [here](https://mail.python.org/pipermail/python-list/2000-March/055125.html) and adapted it to python3:

``` python
import errno

def force_symlink(file1, file2):
    try:
        os.symlink(file1, file2)
    except OSError as e:
        if e.errno == errno.EEXIST:
            os.remove(file2)
            os.symlink(file1, file2)
    ```

Reply:
Corrected. I really hope, that the python-based command can cope properly with the naming of the FEI training files... The shell command did the way I've constructed them.